Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,208 @@
+         *
+         * @method _assignRole
+         * @protected
+         * @param {EventFacade} e
+         */
+        _assignRole: function (e) {
+            var data = e.target.get('data'),
+                userService = this.get('capi').getUserService(),
+                discoveryService = this.get('capi').getDiscoveryService(),
+                service = this;
+
+            this._assignRoleNotificationStarted(
+                data.roleId, data.roleName, e.selection
+            );
+
+            userService.loadRole(data.roleId, function (error, role) {
+                var tasks = new Y.Parallel(),
+                    end = tasks.add(function (err, response) {
+                        if (err) {
+                            //service._error('tasks.add() failed: ' + err.message);
+                            console.log('tasks.add() failed: ', err.message);
+                            return;
+                        }
+
+                        console.log('end'); // TODO just to let us know that it was picked, error handling should be here
+                    });
+
+                if (error) {
+                    service._error('loadRole failed');
+                    return;
+                }
+
+                Y.Array.each(e.selection, function (struct) {
+                    // TODO assign with limitation
+                    //var limitation = {};
+                    var limitation = { // DUMMY DATA
+                        "_identifier": "Section",
+                        "values": {
+                            "ref": [{
+                                "_href": "/api/ezp/v2/content/sections/1",
+                                "_media-type": "application/vnd.ez.api.Section+json"
+                            }, {
+                                "_href": "/api/ezp/v2/content/sections/4",
+                                "_media-type": "application/vnd.ez.api.Section+json"
+                            }]
+                        }
+                    };
+                    var roleAssignInputStruct = userService.newRoleAssignInputStruct(role, limitation),
+                        contentTypeIdentifier = struct.contentType.get('identifier'),
+                        contentIdStr = struct.content.get('id');
+                    //--- TODO we should use another models I think (not content)
+                    //--- so when using instance of such model we can acquire userGroupId and userId
+
+                    if (contentTypeIdentifier === 'user') {
+                        // TODO get this outside of loop
+                        discoveryService.getInfoObject('users', function (error, users) {
+                            if (error) {
+                                service._error('getInfoObject for "users" failed');
+                                return;
+                            }
+
+                            var userIdStr = users._href + '/' + /\d+$/.exec(contentIdStr);
```

Comment:
According to previous comment (https://github.com/ezsystems/PlatformUIBundle/pull/344/files#r40662570) here we don't need to trim the content ID:
```javascript
var userIdStr = users._href + '/' + contentIdStr;
```